jsdom-nodejs-env/src/main/scala/org/scalajs/jsenv/jsdomnodejs/JSDOMNodeJSEnv.scala

```diff
@@ -8,138 +8,225 @@
 
 package org.scalajs.jsenv.jsdomnodejs
 
+import scala.annotation.tailrec
+
 import scala.collection.immutable
+import scala.util.control.NonFatal
 
-import java.io.OutputStream
+import java.io._
+import java.nio.file.{Files, StandardCopyOption}
+import java.net.URI
 
 import org.scalajs.io._
 import org.scalajs.io.JSUtils.escapeJS
 
 import org.scalajs.jsenv._
-import org.scalajs.jsenv.nodejs.AbstractNodeJSEnv
+import org.scalajs.jsenv.nodejs._
 
-class JSDOMNodeJSEnv(config: JSDOMNodeJSEnv.Config) extends AbstractNodeJSEnv {
+class JSDOMNodeJSEnv(config: JSDOMNodeJSEnv.Config) extends JSEnv {
 
   def this() = this(JSDOMNodeJSEnv.Config())
 
-  protected def vmName: String = "Node.js with JSDOM"
+  val name: String = "Node.js with JSDOM"
+
+  def start(input: Input, runConfig: RunConfig): JSRun = {
+    JSDOMNodeJSEnv.validator.validate(runConfig)
+    try {
+      internalStart(initFiles ++ codeWithJSDOMContext(input), runConfig)
+    } catch {
+      case NonFatal(t) =>
+        JSRun.failed(t)
 
-  protected def executable: String = config.executable
+      case t: NotImplementedError =>
+        /* In Scala 2.10.x, NotImplementedError was considered fatal.
+         * We need this case for the conformance tests to pass on 2.10.
+         */
+        JSRun.failed(t)
```

Comment:
Given https://github.com/scala/scala/commit/68f62d7e9c200034bd42ffaf795b57fb379d9d38, perhaps the conformance tests should use a different exception than `NotImplementedError`. Anything else that would be considered `NonFatal` by 2.10.x, really.